src/Elm/Ast.hs

```diff
@@ -79,25 +81,35 @@ data ElmPrim
     | ElmInt                                -- ^ @Int@
     | ElmFloat                              -- ^ @Float@
     | ElmString                             -- ^ @String@
-    | ElmTime                               -- ^ @Posix@ in elm, @UTCTime@ in Haskell
-    | ElmValue                              -- ^ @Json.Encode.Value@ in elm, @Data.Aeson.Value@ in Haskell
-    | ElmMaybe !TypeRef                     -- ^ @Maybe T@
-    | ElmResult !TypeRef !TypeRef           -- ^ @Result A B@ in elm
     | ElmPair !TypeRef !TypeRef             -- ^ @(A, B)@ in elm
     | ElmTriple !TypeRef !TypeRef !TypeRef  -- ^ @(A, B, C)@ in elm
-    | ElmList !TypeRef                      -- ^ @List A@ in elm
-    | ElmNonEmptyPair !TypeRef              -- ^ @NonEmpty A@ represented by @(A, List A)@ in elm
     deriving (Show)
 
+-- | Buitin types defined by core or 3rd party libraries
+-- Included definitions:
+-- * @Maybe a@
+-- * @Result a b@
+-- * @List a@
+-- * @Time.Posix@
+-- * @Json.Encode.Value@
+data ElmBuiltin = ElmBuiltin
+    { builtinImplType    :: !Text
```

Comment:
Also I find the word "Builtin" a bit misleading.
It's feels to me to have the same meaning as "Elm primitive".
How about something like ElmLibrary / ElmLibType ?

Reply:
Impl is leftover from development. I did the change incrementally by porting support from simpler to more complicated types to new structure it will need to be removed.

> It's feels to me to have the same meaning as "Elm primitive" as in "it's provided by Elm itself".

agree though even this is already improvement over including these things into primitives. The one way in which these things are built in even if they come from 3rd party library (say time which depends on `elm-iso8601-date-strings` package) is that their decoders and definition is not generated by elm-street. Including Lib into the name is also option. Though this can be even defined in modules other than the ones generated by elm street as well. This might need some more thinking perhaps.